### PR TITLE
feat!: Commands refactor to make it more user friendly

### LINKED ISSUE #296: [Task] Split dev and terminal, and remove build logs
## What should be done 
In order to make it more readable by the users, remove the Docker build and start logs and use a loading lib (like [loading](https://crates.io/crates/loading)) to display the stages. Make the logs visible with a verbose mode.

## Technical recommandation
Add a verbose mode to the corresponding commands.

## Ecological concerns
Throw the commands outputs only when the verbose mode is on.

### PR DESCRIPTION
**Pull request by taorepoara**
Closes #296 

## Description of the changes
Split the dev mode and the terminal. They don't need to be together anymore.

Don't display the git and Docker commands stdio by default. Use a verbose mode to display them.
Display loading tasks instead.

## Checklist
- [x] I didn't over-scope my PR
- [x] My PR title matches the [commit convention](https://www.conventionalcommits.org/en/v1.0.0/)
- [ ] I did not include breaking changes
- [x] I made my own code-review before requesting one

### I included unit tests that cover my changes
  - [x] 👍 yes
  - [ ] 🙅 no, because they aren't needed
  - [ ] 🙋 no, because I need help

### I added/updated the documentation about my changes

- [ ] 📜 README.md
- [x] 📕 docs/*.md
- [x] 📓 docs.lenra.io
- [ ] 🙅 no documentation needed

### CODE REVIEW

File: docs/commands/index.md [hunk 1]
```diff
@@ -22,21 +24,39 @@ lenra_cli 0.0.0
...
+    build       Build your app in release mode
+    check       Checks the running app
+    dev         Start the app in an interactive mode
+    help        Print this message or the help of the given subcommand(s)
+    logs        View output from the containers
+    new         Create a new Lenra app project from a template
+    reload      Reload the app by rebuilding and restarting it
+    start       Start your app previously built with the build command
+    stop        Stop your app previously started with the start command
+    terminal    Start a Lenra command terminal to run commands with a same context (verbose,
```
Comment (shiipou) [inline docs/commands/index.md:46]: I'm not fan of the name `terminal` to launch this.

I think more natural to start the interactive mode when you just run `lenra` without any sub commands.
Currently it show the help, which is a subcommand `lenra help`.

What did you think about this @taorepoara ?

File: docs/commands/build.md [hunk 2]
```diff
@@ -17,4 +17,6 @@ OPTIONS:
         --config <CONFIG>    The app configuration file [default: lenra.yml]
         --expose <EXPOSE>    Exposes services ports [possible values: app, devtool, postgres, mongo]
     -h, --help               Print help information
+        --production         Remove debug access to the app
```
Comment (shiipou) [inline docs/commands/build.md:20]: Maybe we must document here the alias `--prod` ?

**Review by shiipou — CHANGES_REQUESTED** [review]

File: docs/commands/index.md (hunk 1 above)
Comment (taorepoara) [reply docs/commands/index.md]: I have changed to follow this way

File: docs/commands/new.md [hunk 3]
```diff
@@ -22,6 +22,9 @@ ARGS:
                    You also can set the template project full url to use custom ones
 
 OPTIONS:
-    -h, --help           Print help information
-    -p, --path <PATH>    The new project path [default: .]
+        --config <CONFIG>    The app configuration file [default: lenra.yml]
+        --expose <EXPOSE>    Exposes services ports [possible values: app, devtool, postgres, mongo]
```
[the remark this answers is not in the document]
Comment (taorepoara) [reply docs/commands/new.md:26]: Those options are global, we can't remove them...
Maybe when we'll upgrade to clap 4

File: docs/commands/build.md (hunk 2 above)
Comment (taorepoara) [reply docs/commands/build.md:20]: Seems to be a clap problem

**Pull request merged**